module/zfs/dmu_send.c

```diff
@@ -722,6 +844,17 @@ do_dump(dmu_sendarg_t *dsa, struct send_block_record *data)
 		    !split_large_blocks && !BP_SHOULD_BYTESWAP(bp) &&
 		    !BP_IS_EMBEDDED(bp) && !DMU_OT_IS_METADATA(BP_GET_TYPE(bp));
 
+		/*
+		 * Raw sends only apply to protected blocks. Raw sends are
+		 * mutually exclusive with splitting large blocks and
+		 * compressed sends, so we assert that here.
+		 */
+		boolean_t request_raw =
+		    (dsa->dsa_featureflags & DMU_BACKUP_FEATURE_RAW) != 0;
+
+		IMPLY(request_raw, !request_compressed);
```

Comment:
This is a subjective choice I made, and I would like to hear your thoughts on it as well. Basically I wanted the user to understand the security differences between raw and compressed sends and force them to pick one or the other. At the time this made sense, but I can change it if it doesnt anymore.